File: Map/Map/Map/expression/ArithmeticExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;

namespace Map.expression
{
    class ArithmeticExpression : IExpression
    {
        private char oper;
        private IExpression ex1;
        private IExpression ex2;

        public ArithmeticExpression(char o,IExpression e1,IExpression e2)
        {
            oper = o;
            ex1 = e1;
            ex2 = e2;
        }

        public int eval(IDict<string, int> symbolTable)
        {
            int left = ex1.eval(symbolTable);
            int right = ex2.eval(symbolTable);
            if (oper == '+')
                return left + right;
            if (oper == '-')
			return left - right;
            if (oper == '*')
			return left * right;
            if (oper == '/')
		{
                if (right == 0)
                    throw new Exception("Division by 0!");
                return left / right;
            }
            throw new Exception("Wrong operator!");
        }

        public override string ToString()
        {
            return ex1.ToString() + oper + ex2.ToString();
        }
    }
}

File: Map/Z_InterpretorWCDL/src/expression/ArithmeticExpression.java
package expression;

import domain.DomainException;
import domain.IDictionary;
import domain.IHeap;

public class ArithmeticExpression implements IExpression
{
	private char operator;
	private IExpression left,right;
	
	public ArithmeticExpression(char op, IExpression leftValue, IExpression rightValue)
	{
		operator=op;
		left=leftValue;
		right=rightValue;
	}

	@Override
	public int eval(IDictionary<String, Integer> symbolTable, IHeap<Integer,Integer> heap)
	{
		int left = this.left.eval(symbolTable,heap);
		int right = this.right.eval(symbolTable,heap);
		if(operator=='+')
			return left+right;
		if(operator=='-')
			return left-right;
		if(operator=='*')
			return left*right;
		if(operator=='/')
		{
			if(right==0)
				throw new DomainException("Division by 0!");
			return left/right;
		}
		throw new DomainException("Wrong operator!");
	}
	
	@Override
	public String toString()
	{
		return left.toString()+operator+right.toString();
	}

}
